src/staticanalysis/bot/static_analysis_bot/workflows/local.py

```diff
@@ -64,16 +67,42 @@ def clone(self):
                                     purge=True,
                                     sharebase=settings.repo_shared_dir,
                                     branch=b'central')
-
         cmd.insert(0, hglib.HGPATH)
-        proc = hglib.util.popen(cmd)
+
+        # Start process
+        start = time.time()
+        proc = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+
+        # Set process stdout as non blocking
```

Comment:
Why does it need to be non-blocking? We are reading it ourselves, so even if it is blocking we are unblocking it by reading, no?

Reply:
I think so in theory, but it's still blocking and i won't get any output until the execution end without this

Reply:
Strange, I thought you would get output while the process executes if you read from its stdout.